### PR TITLE
Add some comments to improve documentation

### PR DESCRIPTION
**Pull request by johnynek**
This adds some comments related to #184 and #188

### CODE REVIEW

**Comment by johnynek** [comment]
@morgen-peschke and @timzaak what do you all think of these minor documentation improvements?

**Review by rossabaker — APPROVED** [review]

**Review by kubukoz — APPROVED** [review]

**Comment by timzaak** [comment]
LGTM

**Comment by nafg** [comment]
Where is "arresting failure" defined?

File: core/shared/src/main/scala/cats/parse/Parser.scala [hunk 1]
```diff
@@ -962,6 +966,11 @@ object Parser {
     *  see @backtrack if you want to do backtracking.
     *
     *  This is the same as parsers.foldLeft(fail)(_.orElse(_))
+    *
+    *  recommended style: oneOf(p1 :: p2 :: p3 :: Nil) rather than
```
Comment (nafg) [inline core/shared/src/main/scala/cats/parse/Parser.scala:970]: Why do you care how people define their Lists?

Comment (johnynek) [reply core/shared/src/main/scala/cats/parse/Parser.scala:970]: Using List(a, b, c) uses a var args constructor which allocates and then builds a list after. Using this style avoids that.

IMO it looks nicer to not have the List(..) present as well.

That is why I recommend it.

**Comment by johnynek** [comment]
arresting failure is defined in the comments on the Parser.scala file:

https://github.com/typelevel/cats-parse/blob/65c79bddfd300dd13717525dbc6ba05d5fa70dda/core/shared/src/main/scala/cats/parse/Parser.scala#L45

**Comment by morgen-peschke** [comment]
Looks good 👍

**Pull request merged**